src/tests/util/test_database.rs

```diff
@@ -0,0 +1,158 @@
+use crate::env;
+use diesel::prelude::*;
+use diesel::r2d2::{ConnectionManager, Pool, PooledConnection};
+use diesel::sql_query;
+use diesel_migrations::{FileBasedMigrations, MigrationHarness};
+use once_cell::sync::Lazy;
+use rand::Rng;
+use tracing::instrument;
+use url::Url;
+
+struct TemplateDatabase {
+    base_url: Url,
+    pool: Pool<ConnectionManager<PgConnection>>,
+    template_name: String,
+    prefix: String,
+}
+
+impl TemplateDatabase {
+    #[instrument]
+    pub fn instance() -> &'static Self {
+        static INSTANCE: Lazy<TemplateDatabase> = Lazy::new(TemplateDatabase::new);
+        &INSTANCE
+    }
+
+    #[instrument]
+    fn new() -> Self {
+        let base_url = env("TEST_DATABASE_URL");
+        let base_url = Url::parse(&base_url).expect("failed to parse TEST_DATABASE_URL");
+
+        let prefix = base_url.path().strip_prefix('/');
+        let prefix = prefix.expect("failed to parse database name").to_string();
```

Comment:
this is currently unconditionally using the `TEST_DATABASE_URL` as the management connection URL and also as the prefix for the generated template and test databases. I've kept it like this for now for backwards compatibility purposes. We could potentially also think about using `DATABASE_URL` as the management connection and have a separate `TEST_DATABASE_PREFIX` option, but I think it's sufficient to do that in a follow-up PR if we decide to do so.

Reply:
I think this is fine, but maybe we document this in `docs/CONTRIBUTING.md` as well.

Reply:
✅